src/repart/repart.c

```diff
@@ -9829,7 +9829,9 @@ static int run(int argc, char *argv[]) {
         r = context_open_copy_block_paths(
                         context,
                         loop_device ? loop_device->devno :         /* if --image= is specified, only allow partitions on the loopback device */
-                                      arg_root && !arg_image ? 0 : /* if --root= is specified, don't accept any block device */
+                                      /* if --root= is specified, don't accept any block device, unless it
+                                       * was set automatically because we are in the initrd  */
+                                      arg_root && !arg_image && !PATH_IN_SET(arg_root, "/sysusr/", "/sysroot/") ? 0 :
```

Comment:
i'd prefer if there was an explicit global triste variable `arg_relax_copy_block_security` or so, and that it is set to false in parse_argv() in the --image= handling, but set to true in the automatic initialization of arg_image in the initrd.

(and maybe one day we can even make that flag explicitly configurable, maybe via an env var)